src/sources/socket/udp.rs

```diff
@@ -224,28 +227,28 @@ pub(super) fn udp(
                                             now,
                                         );
 
-                                        let host_key_path = config.host_key.as_ref().map_or_else(
-                                            || [BorrowedSegment::from(log_schema().host_key())],
-                                            |key| [BorrowedSegment::from(key)],
+                                        let legacy_host_key = config.host_key.as_ref().map_or_else(
+                                            || parse_value_path(log_schema().host_key()).ok(),
+                                            |k| k.path.clone(),
                                         );
 
                                         log_namespace.insert_source_metadata(
                                             SocketConfig::NAME,
                                             log,
-                                            Some(LegacyKey::InsertIfEmpty(&host_key_path)),
+                                            legacy_host_key.as_ref().map(LegacyKey::InsertIfEmpty),
                                             path!("host"),
                                             address.ip().to_string()
                                         );
 
-                                        let port_key_path = config.port_key.as_ref().map_or_else(
-                                            || [BorrowedSegment::from("port")],
-                                            |key| [BorrowedSegment::from(key)],
-                                        );
+                                        let legacy_port_key = config
+                                            .port_key
+                                            .as_ref()
+                                            .map_or_else(|| parse_value_path("path").ok(), |k| k.path.clone());
```

Comment:
also here

Reply:
Fixed in [17c7645](https://github.com/vectordotdev/vector/pull/15666/commits/17c76453a258342a101933f934fe8c7c1af5401f)